src/core/transaction_hash/starknet_transaction_hash.rs

```diff
@@ -80,15 +81,13 @@ pub fn calculate_deploy_transaction_hash(
     constructor_calldata: &[Felt],
     chain_id: Felt,
 ) -> Result<Felt, SyscallHandlerError> {
-    let entry_point_selector = CONSTRUCTOR_ENTRY_POINT_SELECTOR.clone();
-
     calculate_transaction_hash_common(
         TransactionHashPrefix::Deploy,
         version,
         contract_address,
-        entry_point_selector,
-        constructor_calldata,
+        CONSTRUCTOR_ENTRY_POINT_SELECTOR.clone(),
         // Field max_fee is considered 0 for Deploy transaction hash calculation purposes.
+        constructor_calldata,
         0,
```

Comment:
The `constructor_calldata` should be before the comment. Maybe we could inline it so this doesn't happen again:
```suggestion
        constructor_calldata,
        0, // Considered 0 for Deploy transaction hash calculation purposes.
```